### PR TITLE
[coordinated] Upgrade react

### PR DESCRIPTION
**Pull request by ds300**
I need hoooks.

Also updated the reaction patch file name to make patch-package 'version mismatch' warning go away.

coordinated with

- artsy/volt#3449
- artsy/reaction#2033
- artsy/palette#296

### CODE REVIEW

**Comment by damassi** [comment]
@ds300 - This should to be a coordinated upgrade across Force and our other JS libs so there aren't version mismatches downstream. If you don't mind tackling that then i'm 👍

File: package.json [hunk 1]
```diff
@@ -183,9 +184,9 @@
     "raven": "^2.6.4",
     "raven-js": "^3.27.0",
     "rc-slider": "^7.0.8",
-    "react": "^16.5.0",
+    "react": "^16.8.2",
```
Comment (damassi) [inline package.json]: Do we intend on upgrading `@types/react` in this PR too? When I was poking at this a few months back during the alpha release it seemed a bit messy, but that could have been resolved in some way.

Comment (ds300) [reply package.json]: yeah, done! I forgot that people write typescript in force.

**Comment by ds300** [comment]
> @ds300 - This should to be a coordinated upgrade across Force and our other JS libs so there aren't version mismatches downstream. If you don't mind tackling that then i'm 👍

Ah I suspected that might need to happen for some reason. Can you explain why? Naïvely speaking, as long as React is backwards-compatible we should be OK to do this in a waterfall fashion rather than all at once.

**Comment by damassi** [comment]
It doesn't explicitly _need_ to be done, but it's more of a team tidiness sort of thing that leads to less confusion with having to keep track of what version of react is supported where.

**Comment by ds300** [comment]
ok cool I'll get on that tomorrow.

**Comment by ds300** [comment]
@damassi ok I made PRs for volt + reaction + palette. Any others you feel are worth doing now?

**Comment by damassi** [comment]
This seems good for now. 

@anandaroop - i'll let you handle the PR to Volt.
@eessex - Heads up re Positron!

**Pull request merged**